components/LocationList.js

```diff
@@ -1,22 +1,71 @@
 import Card from "./Card";
-import styled from "styled-components";
-import Link from "next/link";
-import { StyledSection } from "./StyledSection";
+import { StyledSection, StyledRowSection } from "./StyledSections";
+import { StyledEditButton } from "./StyledButtons";
+import useLocalStorageState from "use-local-storage-state";
+import { FormContainer, Label, Input } from "./StyledFormElements";
+import { StyledEditModal } from "./StyledEditModal";
+import { StyledLink } from "./StyledLinks";
 
-export const StyledLink = styled(Link)`
-  text-decoration: none;
-`;
+export default function LocationList({ locations, setLocations }) {
+  const [showModal, setShowModal] = useLocalStorageState("showModal", {
+    defaultValue: false,
+  });
+
+  const [editingLocation, setEditingLocation] = useLocalStorageState(
+    "editingLocation",
+    { defaultValue: null }
+  );
+
+  function handleEditClick(location) {
+    setEditingLocation(location);
+    setShowModal(true);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const updatedLocations = [...locations];
+
+    const index = updatedLocations.findIndex(
+      (location) => location.id === editingLocation.id
+    );
+
+    updatedLocations[index].locationName = event.target.locationName.value;
+
+    setLocations(updatedLocations);
+
+    setShowModal(false);
+  }
 
-export default function LocationList({ locations }) {
   return (
     <StyledSection>
       {locations.map((location) => {
         return (
-          <StyledLink href={`/locationdetail/${location.id}`} key={location.id}>
-            <Card text={location?.locationName} />
-          </StyledLink>
+          <StyledRowSection key={location.id}>
+            <StyledLink href={`/locationdetail/${location.id}`}>
+              <Card text={location?.locationName} />
+            </StyledLink>
+            <StyledEditButton onClick={() => handleEditClick(location)} />
```

Comment:
you only need the id, so you could also pass it here directly:

```suggestion
            <StyledEditButton onClick={() => handleEditClick(location.id)} />
```

Reply:
I think I do need the location (respectively the colony) here. Otherwise, when I edit, the name of the location/colony is not contained in the edit field (which I want)